Remove confirmation from contractor for one-off payments

Kavitha0809 opened the pull request:
fixes  #1308 

### AI Disclosure
No AI used

### Description
1. Contractors no longer need to accept the payment for one off payments
2. Admin does not need to select the equity share each time

### Demo

1. For contractor with equity enabled

https://github.com/user-attachments/assets/00c92133-5a77-42ea-a6c7-50be36faaecf

2. For contractor without equity enabled

https://github.com/user-attachments/assets/23b557ab-1ada-4a1d-8f6a-dc94b2a1d43d

Kavitha0809 commented on e2e/tests/company/invoices/one-off-payments.spec.ts:
```diff
@@ -94,38 +95,21 @@ test.describe("One-off payments", () => {
-          { page, assertClosed: false },
-        );
-      });
+      test("uses the contractor's configured equity percentage", async ({ page, sentEmails }) => {
```
Tests that the contractors configured equity percentage is visible to admins while making one off payments

Kavitha0809 commented on e2e/tests/company/invoices/one-off-payments.spec.ts:
```diff
@@ -149,142 +134,40 @@ test.describe("One-off payments", () => {
         ]);
       });
 
-      test("with an allowed equity percentage range", async ({ page, sentEmails }) => {
```
This test is no longer required as the admin does not select any equity percentage

Kavitha0809 commented on frontend/app/(dashboard)/invoices/[id]/page.tsx:
```diff
@@ -120,32 +116,6 @@ export default function InvoicePage() {
   const isActionable = useIsActionable();
   const isDeletable = useIsDeletable();
   const isMobile = useIsMobile();
-  const searchParams = useSearchParams();
-  const [acceptPaymentModalOpen, setAcceptPaymentModalOpen] = useState(
-    invoice.requiresAcceptanceByPayee && searchParams.get("accept") === "true",
-  );
-  const acceptPayment = trpc.invoices.acceptPayment.useMutation();
-  const defaultEquityPercentage = invoice.minAllowedEquityPercentage ?? invoice.equityPercentage;
-  const [equityPercentage, setEquityPercentageElected] = useState(defaultEquityPercentage);
-
-  const equityAmountInCents = useMemo(
-    () => (invoice.totalAmountInUsdCents * BigInt(equityPercentage)) / BigInt(100),
-    [equityPercentage],
-  );
-
-  const cashAmountInCents = useMemo(() => invoice.totalAmountInUsdCents - equityAmountInCents, [equityAmountInCents]);
-
-  const acceptPaymentMutation = useMutation({
-    mutationFn: async () => {
-      await acceptPayment.mutateAsync({ companyId: company.id, id, equityPercentage });
-      await refetch();
-      setEquityPercentageElected(defaultEquityPercentage);
-      setAcceptPaymentModalOpen(false);
-    },
-    onSettled: () => {
-      acceptPaymentMutation.reset();
-    },
-  });
```
Removed the "accept payment" related logic as the contractor no longer needs to accept the payment

Kavitha0809 commented on frontend/app/(dashboard)/invoices/[id]/page.tsx:
```diff
@@ -566,15 +465,6 @@ export default function InvoicePage() {
           />
         </div>
       ) : null}
-      {isMobile && user.id === invoice.userId ? (
```
Mobile buttons to accept payment has been removed

Kavitha0809 commented on frontend/app/(dashboard)/people/[id]/page.tsx:
```diff
@@ -288,79 +287,22 @@ export default function ContractorPage() {
                     </FormItem>
                   )}
                 />
-                {company.flags.includes("equity") ? (
```
The below form fields are no longer required as the admin does not need to fill the equity related data.

Kavitha0809 commented on frontend/app/(dashboard)/people/[id]/page.tsx:
```diff
@@ -288,79 +287,22 @@ export default function ContractorPage() {
                   <small className="text-red">{issuePaymentForm.formState.errors.root.message}</small>
                 ) : null}
 
-                <small className="text-muted-foreground">
-                  Your'll be able to initiate payment once it has been accepted by the recipient
-                  {company.requiredInvoiceApprovals > 1 ? " and has sufficient approvals" : ""}.
-                </small>
```
Removed this message as the acceptance is not required any more

Kavitha0809 commented on e2e/tests/company/invoices/one-off-payments.spec.ts:
```diff
@@ -114,76 +120,28 @@ test.describe("One-off payments", () => {
         );
-      });
-
-      test("with a fixed equity percentage", async ({ page, sentEmails }) => {
```
This test is no longer required as the admin does not fill the equity for contractors

Kavitha0809 commented on e2e/tests/company/invoices/one-off-payments.spec.ts:
```diff
@@ -192,100 +150,27 @@ test.describe("One-off payments", () => {
         ]);
       });
-
-      test("allows a worker to accept a one-off payment with a fixed equity percentage", async ({ page }) => {
```
This is no longer required as the worker's default equity percentage is used.

Kavitha0809 commented on frontend/trpc/routes/invoices.ts:
```diff
@@ -63,44 +63,10 @@ const getFlexileFeeCents = (totalAmountCents: bigint) => {
   return feeCents > MAX_FLEXILE_FEE_CENTS ? MAX_FLEXILE_FEE_CENTS : feeCents;
 };
 
-const invoiceInputSchema = createInsertSchema(invoiceLineItems)
```
This input schema had fields related to equity and it's min and max validations. The EquityPercentage is no longer passed by caller instead we pick the value directly from the companyWorker object hence this schema is simplified.

Kavitha0809 commented on frontend/trpc/routes/OneOffInvoiceCreated.tsx:
```diff
@@ -98,8 +98,6 @@ const OneOffInvoiceCreated = ({
           </div>
         </div>
       </div>
-
-      <LinkButton href={`${host}/invoices/${invoice.externalId}?accept=true`}>Accept payment</LinkButton>
```
This button is no longer required as contractor does not accept the payment.

Anshgrover23 commented:
@slavingia @Kavitha0809 — just a heads up, I already have PR #1369 open which makes the same set of changes for issue #1308. This PR seems to overlap almost completely with mine.

Kavitha0809 commented on e2e/tests/company/invoices/one-off-payments.spec.ts:
```diff
@@ -99,20 +100,9 @@ test.describe("One-off payments", () => {
-      await expect(page.getByRole("button", { name: "Accept payment" })).toBeDisabled();
       await expect(page.getByText("Missing tax information.")).toBeVisible();
       await page.getByRole("link", { name: "Invoices" }).click();
-      await page.getByRole("link").getByText("provide your legal details").click();
-      await page.getByLabel("Full legal name (must match your ID)").fill("Legal Name");
-      await page.getByLabel("Tax ID").fill("123456789");
-      await page.getByRole("button", { name: "Save changes" }).click();
-      await withinModal(async (modal) => modal.getByRole("button", { name: "Save", exact: true }).click(), { page });
-      await page.getByRole("link", { name: "Back to app" }).click();
-      await page.getByRole("link", { name: "Invoices" }).click();
-      await page.getByRole("link", { name: invoice.invoiceNumber }).click();
-      await page.getByRole("button", { name: "Accept payment" }).click();
-      await withinModal(async (modal) => modal.getByRole("button", { name: "Accept payment" }).click(), { page });
-      await expect(page.getByRole("button", { name: "Accept payment" })).not.toBeVisible();
```
Payee no longer accepts the payment hence filling the tax form is no longer necessary for this test. The invoice being visible and the `provide your legal details` link present on the page should confirm that the admin can create a payment for user without a complete profile

Anshgrover23 commented:
@slavingia @Kavitha0809 This PR can result in an inconsistent state when admins create invoices for contractors with incomplete tax or legal information.

In contrast, my PR #1369 addresses this scenario more robustly by:

* Automatically updating the `bill_from` field on pending invoices once the contractor completes their tax and legal information (via the `update_pending_invoice_bill_from` callback in `UserComplianceInfo`).
* Ensuring E2E tests accurately reflect the correct invoice state.

This approach prevents broken invoice states and maintains consistency across the system.

Kavitha0809 commented:
@slavingia Could you please let me know why was this one closed?

slavingia commented:
> This PR can result in an inconsistent state when admins create invoices for contractors with incomplete tax or legal information as it doesn't handle setting bill_from field.

Kavitha0809 commented:
@slavingia We were already handling that. The `bill_from` field is set while creating the invoice  by the admin. Below is the code snippet and the link to 

<img width="557" height="537" alt="image" src="https://github.com/user-attachments/assets/31c0789a-f888-4022-bf4d-ad556bba3e58" />

Ref : https://github.com/antiwork/flexile/blob/91224f317dd1d1404a0f9937a05d02915f50a331/frontend/trpc/routes/invoices.ts##L167C1-L197C26

As per the recent PR we had updated it to handle when the contractor accepted the payment we updated those details. Ref https://github.com/antiwork/flexile/pull/1372 . But since we no longer expect the contractor to accept payment, we don't have to update it again as the payment starts immediately.

Kavitha0809 commented:
> Also This PR contains no tests to verify the invoices being created.

There are. REF : https://github.com/antiwork/flexile/blob/90ce6d89dbb902a47c6cf62035689865521549df/e2e/tests/company/invoices/one-off-payments.spec.ts#L57
